### PR TITLE
Add the keyboard operation on recent search items

### PR DESCRIPTION
**Pull request by ktmouk**
## Description
- This PR adds the keyboard operation on recent search items, which helps users to access items quickly. I used `StyledArrowKeyNavigation` component to implement this feature.
- Close: https://github.com/outline/outline/issues/4286

## Recording

https://github.com/outline/outline/assets/17381979/b882762d-8b73-4115-a839-31878f6b5cf5

### CODE REVIEW

File: app/scenes/Search/Search.tsx [hunk 1]
```diff
@@ -178,6 +175,20 @@ class Search extends React.Component<Props> {
...
   };
 
+  get firstResultItem() {
+    const linkItems = this.resultListCompositeRef?.querySelectorAll(
+      "[href]"
+    ) as NodeListOf<HTMLAnchorElement>;
+    return linkItems?.[0];
+  }
+
+  get firstRecentSeatchItem() {
```
Comment (tommoor) [inline app/scenes/Search/Search.tsx:185]:
```suggestion
  get firstRecentSearchItem() {
```

File: app/scenes/Search/Search.tsx [hunk 2]
```diff
@@ -128,12 +129,8 @@ class Search extends React.Component<Props> {
         }
       }
 
-      if (this.compositeRef) {
-        const linkItems = this.compositeRef.querySelectorAll(
-          "[href]"
-        ) as NodeListOf<HTMLAnchorElement>;
-        linkItems[0]?.focus();
-      }
+      const firstItem = this.firstResultItem ?? this.firstRecentSeatchItem;
```
Comment (tommoor) [inline app/scenes/Search/Search.tsx:132]:
```suggestion
      const firstItem = this.firstResultItem ?? this.firstRecentSearchItem;
```

File: app/scenes/Search/components/RecentSearches.tsx [hunk 3]
```diff
@@ -3,16 +3,25 @@ import { CloseIcon } from "outline-icons";
...
 import Fade from "~/components/Fade";
 import NudeButton from "~/components/NudeButton";
 import Tooltip from "~/components/Tooltip";
 import useStores from "~/hooks/useStores";
 import { hover } from "~/styles";
 import { searchPath } from "~/utils/routeHelpers";
 
-function RecentSearches() {
+type Props = {
+  onEscape?: (ev: React.KeyboardEvent<HTMLDivElement>) => void;
```
Comment (tommoor) [inline app/scenes/Search/components/RecentSearches.tsx:18]:
```suggestion
  /** Callback when the Escape key is pressed while navigating the list */
  onEscape?: (ev: React.KeyboardEvent<HTMLDivElement>) => void;
```

**Review by tommoor — COMMENTED** [review]
Looks good, just a small spelling mistake – thanks!

File: app/scenes/Search/Search.tsx (hunk 1 above)
Comment (ktmouk) [reply app/scenes/Search/Search.tsx]: I fixed it, thank you!

**Review by tommoor — APPROVED** [review]

**Comment by bmm-alc** [comment]
thanks for implementing this

**Pull request merged**